addons/account_edi/models/account_edi_format.py

```diff
@@ -495,6 +495,11 @@ def search_with_domain(extra_domain):
 
     def _retrieve_product(self, name=None, default_code=None, barcode=None):
         '''Search all products and find one that matches one of the parameters.
+        Use the following priority:
+        1. the barcode
+        2. the code
+        3. the exact name
+        4. ilike the name
```

Comment:
Thanks for updating the docstring :+1:

Nitpick: I think all the `the`s are unnecessary and this makes it clearer we refer to the parameters
```suggestion
        1. barcode
        2. default_code
        3. name (exact match)
        4. name (ilike)
```